src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/wa/SystemTriggerCompleteHearingOutcomeTask.java

```diff
@@ -0,0 +1,83 @@
+import uk.gov.hmcts.sptribs.systemupdate.service.CcdUpdateService;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.elasticsearch.index.query.QueryBuilders.boolQuery;
+import static org.elasticsearch.index.query.QueryBuilders.matchQuery;
+import static org.elasticsearch.index.query.QueryBuilders.rangeQuery;
+import static uk.gov.hmcts.sptribs.systemupdate.event.SystemTriggerCompleteHearingOutcome.SYSTEM_TRIGGER_COMPLETE_HEARING_OUTCOME;
+
+@Component
+@Slf4j
+public class SystemTriggerCompleteHearingOutcomeTask implements Runnable {
+
+    private final CcdSearchService ccdSearchService;
+
+    private final CcdUpdateService ccdUpdateService;
+
+    private final IdamService idamService;
+
+    private final AuthTokenGenerator authTokenGenerator;
+
+    @Autowired
+    public SystemTriggerCompleteHearingOutcomeTask(CcdSearchService ccdSearchService, CcdUpdateService ccdUpdateService,
+                                                   IdamService idamService, AuthTokenGenerator authTokenGenerator) {
+        this.ccdSearchService = ccdSearchService;
+        this.ccdUpdateService = ccdUpdateService;
+        this.idamService = idamService;
+        this.authTokenGenerator = authTokenGenerator;
+    }
+
+    @Override
+    public void run() {
+        final User user = idamService.retrieveSystemUpdateUserDetails();
+        final String serviceAuth = authTokenGenerator.generate();
+
+        try {
+            final BoolQueryBuilder query = boolQuery()
+                .must(matchQuery("state", "AwaitingHearing"))
+                .filter(rangeQuery("data.hearingList.value.date").lte(LocalDate.now()).gte(LocalDate.now()));
+
+            final List<CaseDetails> casesNeedsStitchCollateHearingBundle =
+                ccdSearchService.searchForAllCasesWithQuery(query, user, serviceAuth);
+            log.info("Cases:" + casesNeedsStitchCollateHearingBundle.size());
+            for (final CaseDetails caseDetails : casesNeedsStitchCollateHearingBundle) {
+                triggerSystemTriggerCompleteHearingOutcomeTask(user, serviceAuth, caseDetails);
+            }
+
+            log.info("System trigger complete hearing outcome scheduled task complete.");
+        } catch (final CcdSearchCaseException e) {
+            log.error("System trigger complete hearing outcome schedule task stopped after search error", e);
+        } catch (final CcdConflictException e) {
+            log.info("System trigger complete hearing outcome schedule task stopping "
+                + "due to conflict with another running task"
+            );
+        }
+    }
+
+    private void triggerSystemTriggerCompleteHearingOutcomeTask(User user, String serviceAuth, CaseDetails caseDetails) {
+        try {
+            log.info("Submitting System Trigger Complete Hearing Outcome Event for Case {}", caseDetails.getId());
+            ccdUpdateService.submitEvent(caseDetails.getId(), SYSTEM_TRIGGER_COMPLETE_HEARING_OUTCOME, user, serviceAuth);
```

Comment:
Might be just a small thing, but should it be 'this.ccdUpdateService.submitEvent' instead of 'ccdUpdateService.submitEvent'?

Reply:
This is correct and in line with the other task classes